Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -362,60 +420,81 @@ void FieldProbe::ComputeDiags (int step)
 
                 // this check is here because for m_field_probe_integrate == True, we always compute
                 // but we only write when we truly are in an output interval step
-                if (m_intervals.contains(step+1)) {
-                    for (int ip = 0; ip < np; ip++) {
+                if (m_intervals.contains(step+1))
+                {
+                    for (auto ip=numparticles - np; ip < numparticles; ip++) 
+                    {
+                        amrex::ParticleReal xp, yp, zp;
+                        getPosition(ip, xp, yp, zp);
                         // Fill output array
-                        m_data[ip * noutputs + FieldProbePIdx::Ex] = part_Ex[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Ey] = part_Ey[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Ez] = part_Ez[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Bx] = part_Bx[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::By] = part_By[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::Bz] = part_Bz[ip];
-                        m_data[ip * noutputs + FieldProbePIdx::S] = part_S[ip];
+                        m_data[ip * noutputs + 0] = ip;
+                        m_data[ip * noutputs + 1] = xp;
+                        m_data[ip * noutputs + 2] = yp;
+                        m_data[ip * noutputs + 3] = zp;
+                        m_data[ip * noutputs + FieldProbePIdx::Ex + 4] = part_Ex[ip];
+                        m_data[ip * noutputs + FieldProbePIdx::Ey + 4] = part_Ey[ip];
+                        m_data[ip * noutputs + FieldProbePIdx::Ez + 4] = part_Ez[ip];
+                        m_data[ip * noutputs + FieldProbePIdx::Bx + 4] = part_Bx[ip];
+                        m_data[ip * noutputs + FieldProbePIdx::By + 4] = part_By[ip];
+                        m_data[ip * noutputs + FieldProbePIdx::Bz + 4] = part_Bz[ip];
+                        m_data[ip * noutputs + FieldProbePIdx::S + 4] = part_S[ip];
                     }
-                    /* m_data now contains up-to-date values for:
-                     *  [Ex, Ey, Ez, Bx, By, Bz, and S] */
+                /* m_data now contains up-to-date values for:
+                 *  [i, Rx, Ry, Rz, Ex, Ey, Ez, Bx, By, Bz, and S] */
                 }
-
-                // do we have the one and only probe particle on our MPI rank?
-                if (np > 0)
-                    probe_proc = amrex::ParallelDescriptor::MyProc();
             }
-
         } // end particle iterator loop
+        unsigned long long sum=0; //numparticles in this level (sum from all processors)
+        int m_MPISize = amrex::ParallelDescriptor::NProcs(); 
+        std::vector<long> result(m_MPISize, 0);
+        amrex::ParallelGather::Gather (numparticles, result.data(), amrex::ParallelDescriptor::IOProcessorNumber(), amrex::ParallelDescriptor::Communicator());
```

Comment:
This should be a Reduce or just `m_probe.TotalNumberOfParticles()`.

Discussed code sample on Thursday was:
```C++
std::vector<double> sendbuf(....);
int local_size = sendbuf.size();
int total_size = 0;
MPI_Reduce(&local_size, &total_size, 1, MPI_INT,
           MPI_SUM, amrex::ParallelDescriptor::IOProcessorNumber(), amrex::ParallelDescriptor::Communicator());
// total_size now has the value of m_probe.TotalNumberOfParticles();
// we can just take auto const total_size = m_probe.TotalNumberOfParticles(); directly

std::vector<double> recvbuf(total_size, 0);
MPI_Gather(sendbuf.data(), send.size(), MPI_DOUBLE,
           recvbuf.data(), recvbuf.size(), MPI_DOUBLE,
           amrex::ParallelDescriptor::IOProcessorNumber(), amrex::ParallelDescriptor::Communicator());
```

### References

MPI Gather:
- low-level understanding:
  - simple form: https://www.mpich.org/static/docs/v3.3/www3/MPI_Gather.html
  - and we need a simple reduce: https://www.mpich.org/static/docs/v3.3/www3/MPI_Reduce.html (or ask the m_probe.TotalNumberOfParticles();)
- What we use is amrex::ParallelGather::Gather https://github.com/AMReX-Codes/amrex/blob/development/Src/Base/AMReX_ParallelReduce.H#L110-L119
  - Here is an example: https://github.com/ECP-WarpX/WarpX/blob/development/Source/Diagnostics/WarpXOpenPMD.cpp#L1231-L1232